nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,401 @@
+    }
+
+    @Test
+    void testMediaContext() {
+        Http1Client client = WebClient.builder()
+                .mediaContext(new CustomizedMediaContext())
+                .build();
+        validateSuccessfulResponse(client);
+    }
+
+    @Test
+    void testChunk() {
+        String[] requestEntityParts = {"First", "Second", "Third"};
+
+        Http1ClientRequest request = getHttp1ClientRequest(Http.Method.PUT, "/test");
+        Http1ClientResponse response = getHttp1ClientResponseFromOutputStream(request, requestEntityParts);
+
+        validateChunkTransfer(response, true, NO_CONTENT_LENGTH, String.join("", requestEntityParts));
+    }
+
+    @Test
+    void testChunkAndChunkResponse() {
+        String[] requestEntityParts = {"First", "Second", "Third"};
+
+        Http1ClientRequest request = getHttp1ClientRequest(Http.Method.PUT, "/chunkresponse");
+        Http1ClientResponse response = getHttp1ClientResponseFromOutputStream(request, requestEntityParts);
+
+        validateChunkTransfer(response, true, NO_CONTENT_LENGTH, String.join("", requestEntityParts));
+        assertThat(response.headers(), hasHeader(Http.HeaderValues.TRANSFER_ENCODING_CHUNKED));
+    }
+
+    @Test
+    void testNoChunk() {
+        String[] requestEntityParts = {"First"};
+        long contentLength = requestEntityParts[0].length();
+
+        Http1ClientRequest request = getHttp1ClientRequest(Http.Method.PUT, "/test")
+                .header(Http.Header.CONTENT_LENGTH, String.valueOf(contentLength));
+        Http1ClientResponse response = getHttp1ClientResponseFromOutputStream(request, requestEntityParts);
+
+        validateChunkTransfer(response, false, contentLength, requestEntityParts[0]);
+    }
+
+    @Test
+    void testForcedChunkNoContentLength() {
+        String[] requestEntityParts = {"First"};
+
+        Http1ClientRequest request = getHttp1ClientRequest(Http.Method.PUT, "/test");
+        Http1ClientResponse response = getHttp1ClientResponseFromOutputStream(request, requestEntityParts);
+
+        validateChunkTransfer(response, true, NO_CONTENT_LENGTH, requestEntityParts[0]);
+    }
+
+    @Test
+    void testForcedChunkTransferEncodingChunked() {
+        String[] requestEntityParts = {"First"};
+
+        Http1ClientRequest request = getHttp1ClientRequest(Http.Method.PUT, "/test")
+                .header(Http.HeaderValues.TRANSFER_ENCODING_CHUNKED);
+        Http1ClientResponse response = getHttp1ClientResponseFromOutputStream(request, requestEntityParts);
+
+        validateChunkTransfer(response, true, NO_CONTENT_LENGTH, requestEntityParts[0]);
+    }
+
+    @Test
+    void testExpect100() {
+        String[] requestEntityParts = {"First", "Second", "Third"};
+
+        Http1Client client = WebClient.builder()
+                .sendExpect100Continue(true)
+                .build();
+        Http1ClientRequest request = client.method(Http.Method.PUT)
+                .uri("http://localhost:" + port + "/test");
```

Comment:
You can inject Http1Client in constructor with the proper URI set in it

Reply:
Done